wordpress-mobile/WordPress-Android
# Handle Quick Start - Task Type List Item action in `ImprovedMySiteFragment`

Issue #15127

This PR opens the `QuickStartFullScreenDialogFragment` from the `ImprovedMySiteFragment` on quick start task type list item click.

To test:

1. Go to My Site -> App Settings -> Test feature configuration.
2. Disable `QuickStartDynamicCardsFeatureConfig`, enable `MySiteImprovementsFeatureConfig`
3. Restart the app
4. Select a site with quick start in progress
5. Click quick start task type list item
6. Notice that `QuickStartFullScreenDialogFragment` gets opened

Out-of-scope: `FullScreenDialogFragment's` `onConfirm`, `onDismiss` action implementation.

## Regression Notes
1. Potential unintended areas of impact
Should have minimal effect as `QuickStartFullScreenDialogFragment` is a separate class, now also opened from the `ImprovedMySiteFragment`

2. What I did to test those areas of impact (or what existing automated tests I relied on)
See "To test" section

3. What automated tests I added (or what prevented me from doing so)
Added unit tests for the newly introduced navigation action `OpenQuickStartFullScreenDialog`

PR submission checklist:
